### PR TITLE
fix(explore): make clicked dnd filters unique

### PR DESCRIPTION
**Pull request by villebro**
### SUMMARY
This applies the fix in #16632 to the DnD filter control.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by kgabryje — DISMISSED** [review]
LGTM!

File: superset-frontend/src/explore/components/DatasourcePanel/types.ts [hunk 1]
```diff
@@ -31,3 +31,7 @@ export function isDatasourcePanelDndItem(
 ): item is DatasourcePanelDndItem {
   return item?.value && item?.type;
 }
+
+export function isDndItemValue(item: any): item is DndItemValue {
+  return item?.value && item?.type;
```
Comment (kgabryje) [inline superset-frontend/src/explore/components/DatasourcePanel/types.ts:36]: Is that correct? `DndItemValue` is `ColumnMeta | Metric`

**Review by kgabryje — APPROVED** [review]
Looks great! 🙂

**Comment by rosemarie-chiu** [comment]
🏷 2021.34

**Pull request merged**